tools/jit/gen_jit_dispatch.py

```diff
@@ -518,6 +572,12 @@ def needs_hacked_twin(decl):
 def hacked_twin(decl):
     decl_copy = copy.deepcopy(decl)
     decl_copy['schema_string'] = decl['schema_string'].replace('Tensor?[]', 'Tensor[]')
+    if decl['overload_name']:
```

Comment:
nit: these lines are kind of long and busy and we're dereferencing stuff out of `decl`/`decl_copy` over and over... if you get the chance before landing, hoisting out some locals should help one and eliminate the other

Reply:
Also, as background, are you fixing a bug here with the overload name dedup? Or is there something about the codegen move that necessitates it?

Reply:
For background: This was a bug but it didn't materialize before